crates/asm-spec/asm.yml

```diff
@@ -358,8 +358,16 @@ StateRead:
           group:
             Sha256:
               opcode: 0x50
-              description: Produce a SHA 256 hash from the specified data.
-              stack_in: [data, data_len]
+              description: |
+                Produce a SHA 256 hash from the specified data.
+                
+                Hashes are byte aligned. Use `extra_padding_len` to remove 
+                padding from the end of the data.
+              panics:
+                - extra_padding_len is greater than 8.
+                - data_len is 0 and padding length is not 0.
+                - data_len is longer than the data.
+              stack_in: [data, data_len, extra_padding_len]
```

Comment:
Are we sure we wanna do this with a 3rd `extra_padding_len` arg, and not just change `data_len` to `data_len_bytes` instead?

Reply:
I left a comment about this above. We could maybe have two ops but I do think that it will be a lot more common to be reading lengths then hashing and all our lengths are in words so we would cause a lot of unnecessary multiply calls

Reply:
I'm guessing if the submitter is submitting a slice of bytes as a dec var, they're probably also submitting the length as another dec var anyways, and submitting the length in bytes is probably easier, rather than one value for words and another value for padding?
Even though our `lengths` are generally in words, you'd always need to know either the byte length in total or the extra padding bytes either way.

Reply:
Yeh I see your point. I think it's hard to know how this plays out. Maybe we can make it byte aligned for now and see how it plays out then we can always add another word aligned op later I guess. We probably need to update the signatures as well.